### PR TITLE
[Merged by Bors] - systest: execute vesting codepath e2e

### LINKED ISSUE #4322: vm, systest: e2e testing for interaction with vesting and vault templates
the e2e test must do the following steps in some capacity:

1. generate public key(s) for vesting wallets. lets do several vesting variants (1/1, 1/2, 2/3, 3/5).
2. generate vault address from vesting wallets and some vault parameters, add them before network initialization. don't setup layer too far from genesis, as test will have too wait for that layer
3. after genesis self-spawn vesting wallet, spawn vault wallet using vesting as a principal
4. drain vault for some number of all layers (5-10), and assert that account balance properly increased

### PR DESCRIPTION
**Pull request by dshulyak**
closes: https://github.com/spacemeshos/go-spacemesh/issues/4322

system tests creates 4 vault accounts and drains them over time

### CODE REVIEW

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

File: systest/tests/smeshing_test.go [hunk 1]
```diff
@@ -142,3 +164,221 @@ func requireEqualEligibilities(tctx *testcontext.Context, tb testing.TB, proposa
...
+				"initial", initial,
+				"current", current,
+				"vested", acc.total,
+				"remaining", remaining,
+				"client", client.Name,
+			)
+			if remaining != 0 {
+				return fmt.Errorf("vault at %v must be empty, instead has %d", acc.vault, remaining)
+			}
+			if delta := int(current - initial); delta+1e7 < acc.total {
```
Comment (countvonzero) [inline systest/tests/smeshing_test.go:262]: - can you show how 1e7 is caclucated
- why isn't this `delta+1e7 != acc.total`?

**Review by countvonzero — APPROVED** [review]

File: systest/tests/smeshing_test.go (hunk 1 above)
Comment (dshulyak) [reply systest/tests/smeshing_test.go:262]: 1e7 is just a small number that should cover all fees. it is not calculated, i just guesstimated it

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

File: systest/tests/smeshing_test.go (hunk 1 above)
Comment (countvonzero) [reply systest/tests/smeshing_test.go:262]: ok. whoever works on this area later will not easily understand this.

**Pull request closed without merging**